### ISSUE TITLE
Bio.Alphabet error & error when loading TCR data from TraCeR

### ISSUE THREAD

**Issue by willblev** [scverse/scirpy#299]
Hi! First off, thanks for developing such a useful package. I have been running into some issues while trying to import data from TraCeR into anndata objects using scirpy following [the Scirpy tutorial](https://icbi-lab.github.io/scirpy/latest/tutorials/tutorial_io.html?highlight=tracer#Read-Smart-seq2-data-processed-with-TraCeR). 

As a quick comment, after creating a fresh Scirpy conda env, I initially got an error about _Bio.Alphabet_ that more recent versions of Biopython stopped including Bio.Alphabet:
```
 File "/path/redacted/anaconda3/envs/scirpy_newest/lib/python3.8/site-packages/scirpy/io/_io.py", line 303, in read_tracer
    tracer_obj = pickle.load(f)
  File "/path/redacted/site-packages/Bio/Alphabet/__init__.py", line 20, in <module>
    raise ImportError(
ImportError: Bio.Alphabet has been removed from Biopython. In many cases, the alphabet can simply be ignored and removed from scripts. In a few cases, you may need to specify the ``molecule_type`` as an annotation on a SeqRecord for your script to work correctly. Please see https://biopython.org/wiki/Alphabet for more information.
```
Rolling back to Biopython version 1.7.2 seemed to solve this. However, when I tried again to import the TCR data from TraCeR using the _io.read_tracer_ function, I get the following error:
```python
import anndata
anndata.logging.anndata_logger.addFilter(
    lambda r: not r.getMessage().startswith("storing")
    and r.getMessage().endswith("as categorical.")
)
import scirpy as ir
import scanpy as sc
from glob import glob
import pandas as pd
import tarfile
import warnings

path_to_tracer_output="path/redacted"
adata_tcr = ir.io.read_tracer(path_to_tracer_output)
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/path/redacted/anaconda3/envs/scirpy/lib/python3.7/site-packages/scirpy/io/_io.py", line 308, in read_tracer
    chains[chain_id], f"TR{chain_id}"
  File "/path/redacted/anaconda3/envs/scirpy/lib/python3.7/site-packages/scirpy/io/_io.py", line 245, in _process_chains
    if tmp_chain.cdr3 == "N/A" or tmp_chain.cdr3nt == "N/A":
AttributeError: 'Recombinant' object has no attribute 'cdr3nt'
```

My Scirpy env is as follows:

- scanpy  1.7.2
- anndata 0.7.6
- scirpy  0.6.1
- pandas  1.1.5
- CPython 3.6.13 (I had to revert to an older version of python to install the older version of Biopython)
- IPython 7.16.1
- Bio 1.72
- numpy 1.19.5 
- scikit-learn 0.24.2 
- statsmodel 0.12.2
- python-igraph 0.9.6

Thank you in advance for your time and if there is anything else you need to know about my setup please ask away!

**Comment by grst** [comment]
Hi @willblev, 

thanks for reporting this issue! 
While I think I can easily find a workaround for the Biopython issue, the second part looks a little more sinister. Essentially, the objects structure of your Tracer output seems different to what scirpy expects. 

What version of tracer are you using? And is there any chance you could send me the Tracer output folder of a single cell that fails? That would be extremely helpful for investigating. 

Best, 
Gregor

**Comment by willblev** [comment]
Thanks for the speedy reply @grst!

> Essentially, the objects structure of your Tracer output seems different to what scirpy expects.

Indeed, this is what I also believed to be the problem.

> What version of tracer are you using?

I downloaded & installed TraCeR a few weeks ago ([following instructions from their GitHub repo](https://github.com/Teichlab/tracer#setup)); the repo says the most recent version is v0.6, however it appears that it actually installs TraCeR v0.5, or at least that is how it shows up in my conda env list. I will look into this further.

>  And is there any chance you could send me the Tracer output folder of a single cell that fails?

I have attached the output of one cell which results in the error. Thanks for looking into this!

[TraCeR_output_example.zip](https://github.com/icbi-lab/scirpy/files/7352993/TraCeR_output_example.zip)
